manticore/core/smtlib/visitors.py

```diff
@@ -301,13 +300,31 @@ def __init__(self, **kw):
         BoolAnd: operator.__and__,
         BoolOr: operator.__or__,
         BoolNot: operator.__not__,
-        BitVecUnsignedDiv: lambda x, y: (x & UNSIGN_MASK) // (y & UNSIGN_MASK),
+        BitVecUnsignedDiv: lambda x, y: 0
+        if (y & UNSIGN_MASK) == 0
+        else (x & UNSIGN_MASK) // (y & UNSIGN_MASK),
         UnsignedLessThan: lambda x, y: (x & UNSIGN_MASK) < (y & UNSIGN_MASK),
         UnsignedLessOrEqual: lambda x, y: (x & UNSIGN_MASK) <= (y & UNSIGN_MASK),
         UnsignedGreaterThan: lambda x, y: (x & UNSIGN_MASK) > (y & UNSIGN_MASK),
         UnsignedGreaterOrEqual: lambda x, y: (x & UNSIGN_MASK) >= (y & UNSIGN_MASK),
     }
 
+    def visit_BitVecDiv(self, expression, *operands):
```

Comment:
```suggestion
    def visit_BitVecDiv(self, expression, *operands) -> BitVecConstant:
```

Reply:
Should not it be Optional[BitVecConstant] ?

Reply:
Ah, yes, you're right